test/test_eager_transforms.py

```diff
@@ -2057,6 +2052,112 @@ def compute_loss(new_params, buffers, x, y):
 
         self.assertEqual(result_grads, expected_grads)
 
+    def test_maml_omniglot_minimal_repro(self, device):
+        # TODO: there appears to be precision issues for float32
+        dtype = torch.double
+
+        n_way = 5
+        n_inner_iter = 2
+        num_tasks = 2
+
+        net = nn.Sequential(
+            nn.Conv2d(1, 64, 3),
+            nn.BatchNorm2d(64, momentum=1, affine=True, track_running_stats=False),
+            nn.Flatten(),
+            nn.Linear(43264, n_way)).to(device).to(dtype)
+
+        fnet, params, buffers = make_functional_with_buffers(net)
+        net = (params, buffers, fnet)
+
+        def loss_for_task(net, n_inner_iter, use_transform, x_spt, y_spt, x_qry, y_qry):
+            params, buffers, fnet = net
+            querysz = x_qry.size(0)
+
+            def compute_loss(new_params, buffers, x, y):
+                logits = fnet(new_params, buffers, x)
+                loss = F.cross_entropy(logits, y)
+                return loss
+
+            new_params = params
+            for _ in range(n_inner_iter):
+                if use_transform:
+                    grads = grad(compute_loss)(new_params, buffers, x_spt, y_spt)
+                else:
+                    res = compute_loss(new_params, buffers, x_spt, y_spt)
+                    grads = torch.autograd.grad(res, new_params, create_graph=True)
+                new_params = [p - g * 1e-1 for p, g, in zip(new_params, grads)]
+
+            qry_logits = fnet(new_params, buffers, x_qry)
+            qry_loss = F.cross_entropy(qry_logits, y_qry)
+            qry_acc = (qry_logits.argmax(
+                dim=1) == y_qry).sum() / querysz
+
+            return qry_loss, qry_acc
+
+        # Get some sample inputs...
+        x_spt = torch.randn(num_tasks, 25, 1, 28, 28, dtype=dtype, device=device)
+        y_spt = torch.randint(0, 5, (num_tasks, 25), device=device)
+        x_qry = torch.randn(num_tasks, 75, 1, 28, 28, dtype=dtype, device=device)
+        y_qry = torch.randint(0, 5, (num_tasks, 75), device=device)
+
+        # compute with vmap + grad
+        compute_loss = partial(loss_for_task, net, n_inner_iter, True)
+        qry_losses, _ = vmap(compute_loss)(x_spt, y_spt, x_qry, y_qry)
+        result_grads = torch.autograd.grad(qry_losses.sum(), params)
+
+        # compute without vmap + grad
+        compute_loss = partial(loss_for_task, net, n_inner_iter, False)
+        losses = [compute_loss(x_spt[i], y_spt[i], x_qry[i], y_qry[i])[0]
+                  for i in range(num_tasks)]
+        expected_grads = torch.autograd.grad(sum(losses), params)
+
+        self.assertEqual(result_grads[2:], expected_grads[2:])  # should pass
+        self.assertEqual(result_grads[:2], expected_grads[:2])  # should fail
+
+    @parametrize('copy', [True, False])
+    @parametrize('originally_track_running_stats', [True, False])
+    def test_update_batch_norm(self, device, copy, originally_track_running_stats):
```

Comment:
This shows that `vmap(grad` works on batch norm, for both cpu and cuda. It's just the math in omniglot that messes it up